From ec2182530da8f787270d4d7dad5213c503215cad Mon Sep 17 00:00:00 2001
From: Alexander Aring <aahringo@redhat.com>
Date: Tue, 21 Mar 2023 09:30:15 -0400
Subject: [PATCH 19/24] dlm_controld: init plocks_data to zero

Inits plocks_data to zero in case of send_all_plocks_data() does not set
plocks_data to any value or nodes_added() returns false. This patch fixes
cppcheck warning:

cpg.c:1299:27: error: Uninitialized variable: plocks_data [uninitvar]
 send_plocks_done(ls, cg, plocks_data);
---
 dlm_controld/cpg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlm_controld/cpg.c b/dlm_controld/cpg.c
index b85fef5f1b45..32aad9246a56 100644
--- a/dlm_controld/cpg.c
+++ b/dlm_controld/cpg.c
@@ -1244,8 +1244,8 @@ static int nodes_added(struct lockspace *ls)
 static void prepare_plocks(struct lockspace *ls)
 {
 	struct change *cg = list_first_entry(&ls->changes, struct change, list);
+	uint32_t plocks_data = 0;
 	struct member *memb;
-	uint32_t plocks_data;
 
 	if (!opt(enable_plock_ind) || ls->disable_plock)
 		return;
-- 
2.35.3

